src/components/Admin/StoriesForApproval.js

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import { Spinner, Card, Media, Badge, Row } from 'reactstrap';
+import _ from 'lodash';
+import '../../styles/TrendingStories.css';
+import axios from 'axios/index';
+import { TRUTHTREE_URI } from '../../constants';
+import { connect } from 'react-redux';
+import ReactHtmlParser from 'react-html-parser';
+
+class StoriesForApproval extends Component {
+  constructor(props) {
+    super(props);
+    this.getStoryDetails = this.getStoryDetails.bind(this);
+    this.selectStory = this.selectStory.bind(this);
+    this.state = {
+      InitialData: [],
+      bgColor: []
+    };
+  }
+
+  contentHtml(data) {
+    return ReactHtmlParser(data);
+  }
+
+  componentDidMount() {
+    //List of stories to be approved if admin
+    axios
+      .get(`${TRUTHTREE_URI}/api/stories/pending`)
+      //Change the api call to unapproved stories
+      .then(response => {
+        let color = [];
+        for (var i = 0; i < response.data.length; i++) {
+          color.push('white');
+        }
+        this.props.dispatch({
+          type: 'STORIES_LIST',
+          adminStories: response.data,
+          adminStoriesLength: response.data.length,
+          bgColor: color,
+          adminSelectedStory: 'none',
+          loading: false
+        });
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }
+
+  //Highlighting the selected story panel & storing its details in redux
+  selectStory(data, index) {
+    let color = [];
+    for (
+      var i = 0;
+      i < this.props.AdminStoriesReducer.adminStoriesLength;
+      i++
+    ) {
+      if (i === index) {
+        color.push('#f2f2f2');
+      } else {
+        color.push('white');
+      }
+    }
+    this.setState({ bgColor: color });
+    this.props.dispatch({
+      type: 'ADMIN_SELECTED_STORY',
+      adminStories: this.props.AdminStoriesReducer.adminStories,
```

Comment:
You can just dispatch the changes and leave out what are intact

Reply:
Added dispatching for only changes, but was unable to access the undispatched variables. Hence, did so